diff -Ndur rebar3-3.1.0/src/rebar_file_utils.erl rebar3-3.1.0-doc/src/rebar_file_utils.erl
--- rebar3-3.1.0/src/rebar_file_utils.erl	2016-04-04 21:41:14.000000000 +0300
+++ rebar3-3.1.0-doc/src/rebar_file_utils.erl	2016-04-10 20:41:48.000000000 +0300
@@ -227,10 +227,10 @@
 
 %% returns an os appropriate tmpdir given a path
 -spec system_tmpdir() -> file:filename().
+system_tmpdir() -> system_tmpdir([]).
+
 -spec system_tmpdir(PathComponents) -> file:filename() when
       PathComponents :: [file:name()].
-
-system_tmpdir() -> system_tmpdir([]).
 system_tmpdir(PathComponents) ->
     Tmp = case erlang:system_info(system_architecture) of
         "win32" ->
