diff -rupN a/tests/bootstrap.php b/tests/bootstrap.php
--- a/tests/bootstrap.php	2015-06-02 17:44:08.000000000 +0200
+++ b/tests/bootstrap.php	2016-06-25 11:01:21.283744068 +0200
@@ -2,19 +2,16 @@
 
 date_default_timezone_set('UTC');
 
-$try = array(
-    __DIR__ . '/../vendor/autoload.php',
-    __DIR__ . '/../../../autoload.php',
-);
+require_once 'Symfony/Component/ClassLoader/ClassLoader.php';
+use Symfony\Component\ClassLoader\ClassLoader;
+$loader = new ClassLoader();
+$loader->setUseIncludePath(true);
+$loader->register();
+$loader->addPrefixes(array(
+	'Sabre' => __DIR__.'/../lib',
+	'SabreTest' => __DIR__,
+));
 
-foreach($try as $path) {
-    if (file_exists($path)) {
-        $autoLoader = include $path;
-        break;
-    }
-}
-
-$autoLoader->addPsr4('Sabre\\VObject\\',__DIR__ . '/VObject');
 
 if (!defined('SABRE_TEMPDIR')) {
   define('SABRE_TEMPDIR', __DIR__ . '/temp/');
