From 945eeaf741bd6b94840adc81acc0281912919a16 Mon Sep 17 00:00:00 2001
From: Aleksander Machniak <machniak@kolabsys.com>
Date: Tue, 1 Aug 2017 13:13:15 +0200
Subject: [PATCH 01/15] Add 'ready' hook for Kolab plugins (Bifrost#T36327)

---
 lib/kolab_sync.php | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/kolab_sync.php b/lib/kolab_sync.php
index 6d3838a..5e9d262 100644
--- a/lib/kolab_sync.php
+++ b/lib/kolab_sync.php
@@ -139,6 +139,8 @@ class kolab_sync extends rcube
             exit;
         }
 
+        $this->plugins->exec_hook('ready', array('task' => 'syncroton'));
+
         // Set log directory per-user
         $this->set_log_dir($this->username ?: $_SERVER['PHP_AUTH_USER']);
 
-- 
2.13.2

