From 661badaa42b1a83cc2430d959d09bf8db4716ee1 Mon Sep 17 00:00:00 2001
From: "Jeroen van Meeuwen (Kolab Systems)" <vanmeeuwen@kolabsys.com>
Date: Fri, 24 Feb 2017 09:34:50 +0100
Subject: [PATCH 2/2] Only call kolab_auth::ldap_close() if the method actually
 exists

---
 lib/kolab_sync.php | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/kolab_sync.php b/lib/kolab_sync.php
index 01bb0c7..150f8ec 100644
--- a/lib/kolab_sync.php
+++ b/lib/kolab_sync.php
@@ -207,7 +207,9 @@ class kolab_sync extends rcube
 
             // Close LDAP connection from kolab_auth plugin
             if (class_exists('kolab_auth', false)) {
-                kolab_auth::ldap_close();
+                if (method_exists('kolab_auth', 'ldap_close')) {
+                    kolab_auth::ldap_close();
+                }
             }
         }
         else {
-- 
2.9.3

