From c2cdb863178c682f544ffbfa9d7d1635c3714de2 Mon Sep 17 00:00:00 2001
From: Aleksander Machniak <machniak@kolabsys.com>
Date: Tue, 20 Jun 2017 10:46:35 +0200
Subject: [PATCH 3/4] Add important note about uid and changed fields in GAL
 config

---
 config/config.inc.php.dist | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/config/config.inc.php.dist b/config/config.inc.php.dist
index 2433475..5e66f59 100644
--- a/config/config.inc.php.dist
+++ b/config/config.inc.php.dist
@@ -54,6 +54,11 @@ $config['activesync_gal_fieldmap'] = null;
 
 // List of device types that will sync the LDAP addressbook(s) as a normal folder.
 // For devices that do not support GAL searching, e.g. Outlook.
+// Note: To make the LDAP addressbook sources working we need two additional
+//       fields ('uid' and 'changed') specified in the fieldmap array
+//       of the LDAP configuration ('ldap_public' option). For example:
+//          'uid'     => 'nsuniqueid',
+//          'changed' => 'modifytimestamp',
 // Examples:
 //     array('windowsoutlook')  # enable for Oultook only
 //     true                     # enable for all
-- 
2.13.0

