From a80f7c51fbafc12a35fe5b82558b8b517ba100cb Mon Sep 17 00:00:00 2001
From: Aleksander Machniak <alec@alec.pl>
Date: Tue, 27 Jun 2017 13:27:24 +0200
Subject: [PATCH 02/14] Update changelog

---
 CHANGELOG | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CHANGELOG b/CHANGELOG
index 1afa5bce6..9e2caf365 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,6 +1,8 @@
 CHANGELOG Roundcube Webmail
 ===========================
 
+- Password: Fix compatibility with PHP 7+ in cpanel_webmail driver (#5820)
+
 RELEASE 1.3.0
 -------------
 - Update to TinyMCE 4.5.7
-- 
2.13.0

