From dda4ec3dcaf0340c2770cad0b14372012957e74d Mon Sep 17 00:00:00 2001
From: johndoh <roundcube@tehinterweb.co.uk>
Date: Thu, 11 Jan 2018 07:00:23 +0000
Subject: [PATCH 12/12] Update composer reference to endroid/qr-code (#6132)

---
 composer.json-dist | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/composer.json-dist b/composer.json-dist
index e29976c52..ad4aed741 100644
--- a/composer.json-dist
+++ b/composer.json-dist
@@ -23,7 +23,7 @@
         "pear/crypt_gpg": "~1.6.0",
         "pear/net_sieve": "~1.4.0",
         "roundcube/plugin-installer": "~0.1.6",
-        "endroid/qrcode": "~1.6.5"
+        "endroid/qr-code": "~1.6.5"
     },
     "require-dev": {
         "phpunit/phpunit": "^4.8.36 || ^5.7.15"
-- 
2.14.3

