From 16e979e9d5e152e913318949ba193eda9a00bb79 Mon Sep 17 00:00:00 2001
From: Aleksander Machniak <alec@alec.pl>
Date: Fri, 30 Jun 2017 11:47:00 +0200
Subject: [PATCH 14/14] Fix c/p issue

---
 program/steps/settings/func.inc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/program/steps/settings/func.inc b/program/steps/settings/func.inc
index 97132bca2..e0be8377c 100644
--- a/program/steps/settings/func.inc
+++ b/program/steps/settings/func.inc
@@ -422,7 +422,7 @@ function rcmail_user_prefs($current = null)
                 $select->add($RCMAIL->gettext('layoutdesktopdesc'), 'desktop');
                 $select->add($RCMAIL->gettext('layoutlistdesc'), 'list');
 
-                $blocks['main']['options']['date_format'] = array(
+                $blocks['main']['options']['layout'] = array(
                     'title'   => html::label($field_id, rcube::Q($RCMAIL->gettext('layout'))),
                     'content' => $select->show($config['layout'] ?: 'widescreen'),
                 );
-- 
2.13.0

