diff --git a/src/riak_ensemble_manager.erl b/src/riak_ensemble_manager.erl
index f0a1085..4cbb7ce 100644
--- a/src/riak_ensemble_manager.erl
+++ b/src/riak_ensemble_manager.erl
@@ -487,7 +487,7 @@ reload_state() ->
 -spec initial_state() -> state().
 initial_state() ->
     ets:insert(?ETS, {enabled, false}),
-    ClusterName = {node(), erlang:now()},
+    ClusterName = {node(), os:timestamp()},
     CS = riak_ensemble_state:new(ClusterName),
     State=#state{version=0,
                  ensemble_data=[],
diff --git a/src/synctree_leveldb.erl b/src/synctree_leveldb.erl
index 69d005d..0deeef9 100644
--- a/src/synctree_leveldb.erl
+++ b/src/synctree_leveldb.erl
@@ -87,7 +87,7 @@ get_path(Opts) ->
     case proplists:get_value(path, Opts) of
         undefined ->
             Base = "/tmp/ST",
-            Name = integer_to_list(timestamp(erlang:now())),
+            Name = integer_to_list(timestamp(os:timestamp())),
             filename:join(Base, Name);
         Path ->
             Path
