From 25fe73820c89e4098d9e415c2b9d57998126c8f0 Mon Sep 17 00:00:00 2001
From: Aleksander Machniak <machniak@kolabsys.com>
Date: Thu, 11 Jan 2018 08:15:41 +0100
Subject: [PATCH 1/3] Remove activesync_user_debug also from sample config file

---
 config/config.inc.php.dist | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/config/config.inc.php.dist b/config/config.inc.php.dist
index 61a4fd8..1b9b011 100644
--- a/config/config.inc.php.dist
+++ b/config/config.inc.php.dist
@@ -8,9 +8,6 @@ $config['activesync_debug'] = true;
 // Enables logging to a separate directory for every user/device
 $config['activesync_user_log'] = false;
 
-// Enable per-user debugging only if /var/log/kolab-syncroton/<username>/ folder exists
-$config['activesync_user_debug'] = false;
-
 // If specified all ActiveSync-related logs will be saved to this file
 // Note: This doesn't change Roundcube Framework log locations
 $config['activesync_log_file'] = null;
-- 
2.14.3

