From 764c71d5df5786ed5b74d11a4904b282b5a29c4a Mon Sep 17 00:00:00 2001
From: "Jeroen van Meeuwen (Kolab Systems)" <vanmeeuwen@kolabsys.com>
Date: Mon, 18 May 2015 11:48:22 +0200
Subject: [PATCH 4/7] Allow discarding building ports

---
 src/rebar_port_compiler.erl | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/rebar_port_compiler.erl b/src/rebar_port_compiler.erl
index 35adc3c..04a4382 100644
--- a/src/rebar_port_compiler.erl
+++ b/src/rebar_port_compiler.erl
@@ -278,6 +278,12 @@ get_specs(Config, AppFile) ->
                     %% No spec provided. Construct a spec
                     %% from old-school so_name and sources
                     [port_spec_from_legacy(Config, AppFile)];
+                [{null,[]}] ->
+                    [];
+                [null] ->
+                    [];
+                [skip] ->
+                    [];
                 PortSpecs ->
                     Filtered = filter_port_specs(PortSpecs),
                     OsType = os:type(),
-- 
2.1.0

