diff --git a/src/riak_pipe.erl b/src/riak_pipe.erl
index 9f3ec10..81031b2 100644
--- a/src/riak_pipe.erl
+++ b/src/riak_pipe.erl
@@ -680,7 +680,7 @@ example_tick(TickLen, BatchSize, NumTicks, ChainLen) ->
     {ok, Pipe} = riak_pipe:exec(Specs, [{log, sink},
                                         {trace, all}]),
     _ = [begin
-             _ = [ok = riak_pipe:queue_work(Pipe, {tick, {TickSeq, X}, now()})
+             _ = [ok = riak_pipe:queue_work(Pipe, {tick, {TickSeq, X}, os:timestamp()})
                   || X <- lists:seq(1, BatchSize)],
              if TickSeq /= NumTicks -> timer:sleep(TickLen);
                 true                -> ok
diff --git a/src/riak_pipe_vnode.erl b/src/riak_pipe_vnode.erl
index 8487d98..e65d36d 100644
--- a/src/riak_pipe_vnode.erl
+++ b/src/riak_pipe_vnode.erl
@@ -478,7 +478,7 @@ reply_archive(Pid, Fitting, Archive) ->
 %%      Total time the worker has spent waiting for inputs (as opposed
 %%      to working on them).  Given as an integer number of
 %%      microseconds.  Should be roughly equal to
-%%      `(now()-started)-work_time'.
+%%      `(os:timestamp()-started)-work_time'.
 %%</dd></dl>
 -spec status(pid()) -> {partition(), [[{atom(), term()}]]}.
 status(Pid) ->
