--- roundcubemail-1.2.2.orig/vendor/composer/autoload_psr4.php	2016-11-28 01:19:22.926011543 +0100
+++ roundcubemail-1.2.2/vendor/composer/autoload_psr4.php	2016-11-28 00:42:00.516341512 +0100
@@ -6,6 +6,6 @@
 $baseDir = dirname($vendorDir);
 
 return array(
-    'Sabre\\' => array($baseDir . '/../../../share/php/sabre21'),
-    '' => array($baseDir . '/../../../share/php'),
+    'Sabre\\' => array('/usr/share/php/sabre21'),
+    '' => array('/usr/share/php'),
 );
