--- roundcubemail-1.0.orig/program/lib/Roundcube/rcube_ldap.php	2013-08-09 09:15:51.000000000 +0100
+++ roundcubemail-1.0/program/lib/Roundcube/rcube_ldap.php	2013-09-09 17:04:20.115646526 +0100
@@ -811,7 +811,7 @@
 
         // add general filter to query
         if (!empty($this->prop['filter']))
-            $filter = '(&(' . preg_replace('/^\(|\)$/', '', $this->prop['filter']) . ')' . $filter . ')';
+            $filter = '(&(' . preg_replace('/^\(|\)$/', '', array_key_exists('adv_filter', $this->prop) ? $this->prop['adv_filter'] : $this->prop['filter']) . ')' . $filter . ')';
 
         // set filter string and execute search
         $this->set_search_set($filter);
