diff -ur basho-cuttlefish-f3645f6-relax-lager-dep/src/cuttlefish.app.src basho-cuttlefish-f3645f6/src/cuttlefish.app.src
--- basho-cuttlefish-f3645f6-relax-lager-dep/src/cuttlefish.app.src	2015-06-27 02:36:09.000000000 +0200
+++ basho-cuttlefish-f3645f6/src/cuttlefish.app.src	2015-07-08 14:19:44.834855274 +0200
@@ -1,7 +1,7 @@
 {application, cuttlefish,
  [
   {description, "cuttlefish configuration abstraction"},
-  {vsn, git},
+  {vsn, "2.0.3"},
   {registered, []},
   {applications, [
                   kernel,
