diff -ur basho-clique-ce3bdf7.orig/src/clique.app.src basho-clique-ce3bdf7/src/clique.app.src
--- basho-clique-ce3bdf7.orig/src/clique.app.src	2015-06-22 23:24:07.000000000 +0200
+++ basho-clique-ce3bdf7/src/clique.app.src	2015-07-08 16:34:34.343919891 +0200
@@ -1,7 +1,7 @@
 {application, clique,
  [
   {description, "A CLI library for erlang"},
-  {vsn, git},
+  {vsn, "0.3.0"},
   {modules, []},
   {registered, []},
   {applications, [
