From 6769c10da417e65698f71e0f34d0b15a597a3641 Mon Sep 17 00:00:00 2001
From: "Jeroen van Meeuwen (Kolab Systems)" <vanmeeuwen@kolabsys.com>
Date: Tue, 19 Jan 2016 12:55:44 +0100
Subject: [PATCH 1/2] Fix mistake in manticore configuration template

---
 share/templates/manticore.js.tpl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/share/templates/manticore.js.tpl b/share/templates/manticore.js.tpl
index 90a75ec..c3a31bd 100644
--- a/share/templates/manticore.js.tpl
+++ b/share/templates/manticore.js.tpl
@@ -17,7 +17,7 @@ module.exports = {
   CHWALA_SERVER: 'http://$fqdn/chwala/api/document',
   ROUNDCUBE_SERVER: 'http://$fqdn/roundcubemail',
 
-  AUTH_ENCRYPTION_KEY: '$auth_secret',
+  AUTH_ENCRYPTION_KEY: '$auth_key',
 
   LDAP_SERVER: 'ldap://$server_host:389',
   LDAP_BASE: '$user_base_dn',
-- 
2.4.3

