From 8e7ae3827770acf071eaea187f017d099d210e95 Mon Sep 17 00:00:00 2001
From: "Jeroen van Meeuwen (Kolab Systems)" <vanmeeuwen@kolabsys.com>
Date: Fri, 8 Jul 2016 13:16:55 +0200
Subject: [PATCH] Call the correct functions

---
 kolabd/__init__.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/kolabd/__init__.py b/kolabd/__init__.py
index 9629836..a5869d5 100644
--- a/kolabd/__init__.py
+++ b/kolabd/__init__.py
@@ -285,7 +285,7 @@ class KolabDaemon(object):
 
             for primary in primaries:
                 naming_context = primary_auth.domain_naming_context(primary)
-                domain_root_dn = primary_auth.domain_root_dn(primary)
+                domain_root_dn = primary_auth._auth._kolab_domain_root_dn(primary)
                 log.debug(
                         _("Domain %r naming context: %r, root dn: %r") % (
                                 primary,
@@ -307,7 +307,6 @@ class KolabDaemon(object):
 
             # Find however many naming contexts we have, and what the
             # corresponding domain name is for them.
-            naming_contexts = list(set(naming_contexts.values()))
             primary_domains = [x for x,y in naming_contexts.iteritems() if domain_root_dns[x] == y]
 
             # Now we can check if any changes happened.
-- 
2.7.4

