diff --git a/pykolab/setup/setup_mta.py b/pykolab/setup/setup_mta.py
index 53293cf..0a13fc3 100644
--- a/pykolab/setup/setup_mta.py
+++ b/pykolab/setup/setup_mta.py
@@ -472,11 +472,11 @@ result_format = "shared+%%s"
         log.error(_("Could not find a ClamAV update configuration file"))
 
     if os.path.isfile('/etc/freshclam.conf'):
-        subprocess.call(
+        subprocess.call([
                 '/usr/bin/freshclam',
                 '--quiet',
                 '--datadir="/var/lib/clamav"'
-            )
+            ])
 
     amavisservice = 'amavisd.service'
     clamavservice = 'clamd@amavisd.service'
