diff -rupN a/tests/bootstrap.php b/tests/bootstrap.php
--- a/tests/bootstrap.php	2016-09-04 17:10:06.666787641 +0200
+++ b/tests/bootstrap.php	2016-09-04 17:12:35.840848334 +0200
@@ -1,17 +1,17 @@
 <?php
 
-set_include_path(__DIR__ . '/../lib/' . PATH_SEPARATOR . __DIR__ . PATH_SEPARATOR . get_include_path());
-
-$autoLoader = include __DIR__ . '/../vendor/autoload.php';
-
-// SabreDAV tests auto loading
-$autoLoader->add('Sabre\\', __DIR__);
-// VObject tests auto loadiong
-$autoLoader->addPsr4('Sabre\\VObject\\',__DIR__ . '/../vendor/sabre/vobject/tests/VObject');
-
-
 date_default_timezone_set('UTC');
 
+require_once 'Symfony/Component/ClassLoader/ClassLoader.php';
+use Symfony\Component\ClassLoader\ClassLoader;
+$loader = new ClassLoader();
+$loader->setUseIncludePath(true);
+$loader->register();
+$loader->addPrefixes(array(
+        'Sabre' => __DIR__.'/../lib',
+        'SabreTest' => __DIR__,
+));
+
 $config = [
     'SABRE_TEMPDIR'   => dirname(__FILE__) . '/temp/',
     'SABRE_HASSQLITE' => in_array('sqlite',PDO::getAvailableDrivers()),
