From: =?utf-8?q?David_Pr=C3=A9vot?= <taffit@debian.org>
Date: Thu, 26 Feb 2015 14:07:53 -0400
Subject: Install sabredav.php in /usr/share/php/sabre21/Sabre

Forwarded: not-needed
---
 bin/sabredav | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/sabredav b/bin/sabredav
index 032371b..956d19b 100755
--- a/bin/sabredav
+++ b/bin/sabredav
@@ -1,2 +1,2 @@
 #!/bin/sh
-php -S 0.0.0.0:8080 `dirname $0`/sabredav.php
+php -S 0.0.0.0:8080 `dirname $0`/../share/php/sabre21/Sabre/sabredav.php
