From 18077fc3c77ab9d62e8691d853e5516a7a765281 Mon Sep 17 00:00:00 2001
From: "Jeroen van Meeuwen (Kolab Systems)" <vanmeeuwen@kolabsys.com>
Date: Fri, 25 Aug 2017 15:17:27 +0200
Subject: [PATCH 1/2] [dlist]: Use int8_t in dlist_parsemap()

Resolves #2119
---
 imap/dlist.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/imap/dlist.c b/imap/dlist.c
index c93b85c..83ae73d 100644
--- a/imap/dlist.c
+++ b/imap/dlist.c
@@ -843,7 +843,7 @@ EXPORTED char dlist_parse(struct dlist **dlp, int parsekey, struct protstream *i
     struct dlist *dl = NULL;
     static struct buf kbuf;
     static struct buf vbuf;
-    int c;
+    int8_t c;
 
     /* handle the key if wanted */
     if (parsekey) {
@@ -957,7 +957,7 @@ EXPORTED int dlist_parsemap(struct dlist **dlp, int parsekey,
 		   const char *base, unsigned len)
 {
     struct protstream *stream;
-    char c;
+    int8_t c;
     struct dlist *dl = NULL;
 
     stream = prot_readmap(base, len);
-- 
1.8.3.1

