diff -ur kolab-schema-3.1.orig/kolab3.schema kolab-schema-3.1/kolab3.schema
--- kolab-schema-3.1.orig/kolab3.schema	2012-09-12 11:45:00.000000000 +0200
+++ kolab-schema-3.1/kolab3.schema	2014-02-12 11:53:58.967583797 +0100
@@ -43,7 +43,6 @@
 # Prefix for objectclasses: 1.3.6.1.4.1.19414.3
 # nameprefix: kolab
 #
-dn: cn=schema
 ####################
 # kolab attributes #
 ####################
@@ -166,6 +165,17 @@
   EQUALITY caseIgnoreIA5Match
   SUBSTR caseIgnoreIA5SubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{256} )
+# Forward all incoming emails except UCE if kolabForwardUCE
+# is not set to this email address.
+# There can be multiple kolabForwardAddress entries for
+# each kolabInetOrgPerson.
+# Currently this attribute is not used in Kolab.
+attributetype ( 1.3.6.1.4.1.19414.1.1.1.14
+  NAME 'kolabForwardAddress'
+  DESC 'Forward email to this address'
+  EQUALITY caseIgnoreIA5Match
+  SUBSTR caseIgnoreIA5SubstringsMatch
+  SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{256} )
 # Keep local copy when forwarding emails to list of
 # kolabForwardAddress.
 # Default is no.
@@ -316,7 +326,6 @@
   MUST cn
   MAY ( acl $
         alias $
-        mailHost $
         kolabFolderType $
         kolabDeleteflag $
         kolabDelegate $
@@ -334,7 +343,6 @@
   DESC 'Kolab Internet Organizational Person'
   SUP top AUXILIARY
   MAY ( alias $
-        mailHost $
         kolabHomeServerOnly $
         kolabDelegate $
         kolabInvitationPolicy $
@@ -344,6 +352,7 @@
         kolabVacationAddress $
         kolabVacationReplyToUCE $
         kolabVacationReactDomain $
+        kolabForwardAddress $
         kolabForwardKeepCopy $
         kolabForwardUCE $
         kolabAllowSMTPRecipient $
