Description: Run the daemon using "dirsrv" user and group
Author: Michele Baldessari <michele@acksyn.org>

Index: 389-ds-base/configure.ac
===================================================================
--- 389-ds-base.orig/configure.ac	2011-09-30 01:18:51.582181551 +0300
+++ 389-ds-base/configure.ac	2011-10-10 11:02:23.765296773 +0300
@@ -289,8 +289,8 @@
 schemadir=/$PACKAGE_NAME/schema
 
 # default user, group
-defaultuser=nobody
-defaultgroup=nobody
+defaultuser=dirsrv
+defaultgroup=dirsrv
 
 if test -z "$with_tmpfiles_d" ; then
    if test -d $sysconfdir/tmpfiles.d ; then
