Index: xapian-core-1fmwheezy28569/queryparser/snippetgenerator_internal.cc
===================================================================
--- xapian-core-1fmwheezy28569.orig/queryparser/snippetgenerator_internal.cc	2013-05-24 07:09:07.641153411 +0000
+++ xapian-core-1fmwheezy28569/queryparser/snippetgenerator_internal.cc	2013-05-24 07:09:12.756884500 +0000
@@ -248,7 +248,9 @@ SnippetGenerator::Internal::accept_text(
 	}
 
 	while (true) {
-	    if (cjk_ngram && CJK::codepoint_is_cjk(*itor)) {
+	    if (cjk_ngram &&
+		CJK::codepoint_is_cjk(*itor) &&
+		Unicode::is_wordchar(*itor)) {
 		const string & cjk = CJK::get_cjk(itor);
 		for (CJKTokenIterator tk(cjk); tk != CJKTokenIterator(); ++tk) {
 		    const string & cjk_token = *tk;
