--- mozilla/directory/java-sdk/ldap.mk
+++ mozilla/directory/java-sdk/ldap.mk
@@ -143,12 +143,12 @@
 ifndef JAVAC
   ifndef DEBUG
      #defualt mode is debug (-g)
-    JAVAC=$(JDKBIN)javac -g -classpath "$(JAVACLASSPATH)"
+    JAVAC=$(JDKBIN)javac -source 1.4 -g -classpath "$(JAVACLASSPATH)"
   else
   ifeq ($(DEBUG), 1)
-    JAVAC=$(JDKBIN)javac -g -classpath "$(JAVACLASSPATH)"
+    JAVAC=$(JDKBIN)javac -source 1.4 -g -classpath "$(JAVACLASSPATH)"
   else
-    JAVAC=$(JDKBIN)javac -O -classpath "$(JAVACLASSPATH)"
+    JAVAC=$(JDKBIN)javac -source 1.4 -O -classpath "$(JAVACLASSPATH)"
   endif
   endif
 endif
@@ -158,7 +158,7 @@
 endif
 
 ifndef JAVADOC
-  JAVADOC=$(JDKBIN)javadoc -classpath "$(JAVACLASSPATH)$(SEP)$(BASEDIR)/ldapbeans"
+  JAVADOC=$(JDKBIN)javadoc -source 1.4 -classpath "$(JAVACLASSPATH)$(SEP)$(BASEDIR)/ldapbeans"
 endif
 
 # Destination for class files and packages
--- mozilla/directory/java-sdk/ldapsp.mk
+++ mozilla/directory/java-sdk/ldapsp.mk
@@ -77,12 +77,12 @@
 ifndef JAVAC
   ifndef DEBUG
      #defualt mode is debug (-g)
-    JAVAC=$(JDKBIN)javac -g -classpath "$(JAVACLASSPATH)"
+    JAVAC=$(JDKBIN)javac -source 1.4 -g -classpath "$(JAVACLASSPATH)"
   else
   ifeq ($(DEBUG), 1)
-    JAVAC=$(JDKBIN)javac -g -classpath "$(JAVACLASSPATH)"
+    JAVAC=$(JDKBIN)javac -source 1.4 -g -classpath "$(JAVACLASSPATH)"
   else
-    JAVAC=$(JDKBIN)javac -O -classpath "$(JAVACLASSPATH)"
+    JAVAC=$(JDKBIN)javac -source 1.4 -O -classpath "$(JAVACLASSPATH)"
   endif
   endif
 endif
@@ -92,7 +92,7 @@
 endif
 
 ifndef JAVADOC
-  JAVADOC=$(JDKBIN)javadoc -classpath "$(JAVACLASSPATH)"
+  JAVADOC=$(JDKBIN)javadoc -source 1.4 -classpath "$(JAVACLASSPATH)"
 endif
 
 all: classes 
