--- 389-ds-base-1.2.11.15/ldap/servers/slapd/ldaputil.c.orig	2012-09-25 18:16:09.000000000 +0200
+++ 389-ds-base-1.2.11.15/ldap/servers/slapd/ldaputil.c	2014-05-08 06:48:04.000000000 +0200
@@ -682,8 +682,16 @@
         if (!(pluginpath = getenv("SASL_PATH"))) {
 #if defined(LINUX) && defined(__LP64__)
             pluginpath = "/usr/lib64/sasl2";
+            if (PR_SUCCESS != PR_Access(pluginpath, PR_ACCESS_EXISTS)) {
+                /* Since the path does not exist, try gnu triplet. */
+                pluginpath = "/usr/lib/x86_64-linux-gnu/sasl2";
+            }
 #else
             pluginpath = "/usr/lib/sasl2";
+            if (PR_SUCCESS != PR_Access(pluginpath, PR_ACCESS_EXISTS)) {
+                /* Since the path does not exist, try gnu triplet. */
+                pluginpath = "/usr/lib/i386-linux-gnu/sasl2";
+            }
 #endif
         }
     }
--- 389-ds-base-1.2.11.15/ldap/servers/slapd/saslbind.c.orig	2012-09-25 18:16:09.000000000 +0200
+++ 389-ds-base-1.2.11.15/ldap/servers/slapd/saslbind.c	2014-05-08 06:48:04.000000000 +0200
@@ -530,8 +530,16 @@
         if (!(pluginpath = getenv("SASL_PATH"))) {
 #if defined(LINUX) && defined(__LP64__)
             pluginpath = "/usr/lib64/sasl2";
+            if (PR_SUCCESS != PR_Access(pluginpath, PR_ACCESS_EXISTS)) {
+                /* Since the path does not exist, try gnu triplet. */
+                pluginpath = "/usr/lib/x86_64-linux-gnu/sasl2";
+            }
 #else
             pluginpath = "/usr/lib/sasl2";
+            if (PR_SUCCESS != PR_Access(pluginpath, PR_ACCESS_EXISTS)) {
+                /* Since the path does not exist, try gnu triplet. */
+                pluginpath = "/usr/lib/i386-linux-gnu/sasl2";
+            }
 #endif
         }
     }
