From 11ca677bada174536fa1f7074892b1aa0f04091a Mon Sep 17 00:00:00 2001
From: "Jeroen van Meeuwen (Kolab Systems)" <vanmeeuwen@kolabsys.com>
Date: Fri, 25 Aug 2017 15:43:35 +0200
Subject: [PATCH 2/2] Avoid returning -1 as 255

Resolves #2122
---
 cunit/parse.testc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cunit/parse.testc b/cunit/parse.testc
index 281b249..dcaf920 100644
--- a/cunit/parse.testc
+++ b/cunit/parse.testc
@@ -23,7 +23,7 @@ int wrap_getint32(const char *s, int32_t *valp)
 {
     struct protstream *prot;
     char *b;
-    int c;
+    int8_t c;
 
     b = xstrdup(s);	/* work around bug in prot_ungetc */
     prot = prot_readmap(b, strlen(b));
@@ -101,7 +101,7 @@ int wrap_getsint32(const char *s, int32_t *valp)
 {
     struct protstream *prot;
     char *b;
-    int c;
+    int8_t c;
 
     b = xstrdup(s);	/* work around bug in prot_ungetc */
     prot = prot_readmap(b, strlen(b));
-- 
1.8.3.1

